### ISSUE TITLE
"ignoreAccessorPattern" doesn't behave like the TSLint version

### ISSUE THREAD

**Issue by ulrichb** [jonaskello/eslint-plugin-functional#44]
My new ESLint config:
```
        "functional/immutable-data": ["error", {
            ignorePattern: ["^mutable"],
            ignoreAccessorPattern: ["**.mutable*.*", "**.current.*"],
        }],
```

My old TSLint config: 

```
"no-object-mutation":
  [true, {"ignore-prefix": "mutable", "ignore-pattern": ["**.mutable*", "**.current"]}]
```

And here the sample code, where it behaves differently:

```TS
export class SomeClass {

    private mutableSomething?: string;

    mutate(): void {
        this.mutableSomething = "xx"; // Ignored in TSLint, now reported
    }
}

//

const somethingLikeReactUseRef = { current: "x" };

somethingLikeReactUseRef.current = "y"; // Ignored in TSLint, now reported
```

Am I doing something wrong?

**Comment by RebeccaStevens** [comment]
Hey, the `ignoreAccessorPattern` option in this project works a bit different to the `ignore-pattern` option from the old tslint version.

The option in the tslint version was designed to work on based on the name of the "container" being modified, while the new eslint version's option is designed to work based on the name of the object being modified.

For example, with `foo.bar = 1;`, bar is the container being modified while foo is the object being modified.

**Comment by ulrichb** [comment]
Hi @RebeccaStevens.

Thanks for your answer.

I've updates my settings to `["**.mutable*.*", "**.current.*"]` (and also the description above).

Still the same issue with the sample code above.

Could you verify this (and reopen if yes)?

**Comment by RebeccaStevens** [comment]
I'll take another look a bit late. I'll reopen to remind myself to do so.

**Comment by RebeccaStevens** [comment]
Hi @ulrichb,

Sorry I got it wrong before, you caught me just as I was going to bed :stuck_out_tongue:.

As of right now there is no way to ignore based on a property name (what I refer to as the "container" earlier), only on an object name.

So with `this.mutableSomething = "xx";`, `this` is name of the object and with `somethingLikeReactUseRef.current = "y";`, `somethingLikeReactUseRef` is the name of the object.

So this config should work:
```json
"functional/immutable-data": ["error", {
  "ignoreAccessorPattern": ["this", "somethingLikeReactUseRef"]
}],
```

Obviously this isn't ideal for your use case so I'll take a look into possible solutions. I'll open a new issue for this.

**Comment by RebeccaStevens** [comment]
#47 should make things work a lot more like in the tslint version.

**Comment by ulrichb** [comment]
Ah. okay. I misunderstood the (current) meaning of `ignoreAccessorPattern`. 

So with #47 you change that behavior (to my original expectation), right?

**Comment by RebeccaStevens** [comment]
Yes,

This config should work for your original code (once the update is merged in):

```json
"functional/immutable-data": ["error", {
  ignoreAccessorPattern: ["**.mutable*", "**.current"],
}]
```

As only string are being modified in your code, property mutation doesn't need to be checked.
However, to allow for reassigning and property mutation use:

```json
"functional/immutable-data": ["error", {
  ignoreAccessorPattern: ["**.mutable*.**", "**.current.**"],
}]
```

**Comment by ulrichb** [comment]
Many, many thanks @RebeccaStevens !

Looking forward to the next beta release to try this out.

**Comment by RebeccaStevens** [comment]
@ulrichb v0.5 is now out

**Comment by ulrichb** [comment]
Just updated. Works great! Many thanks!
